ludwig/features/text_feature.py

```diff
@@ -68,7 +68,7 @@ def get_decoded_targets_and_predictions(
     """Returns the decoded targets and predictions, accounting for IGNORE_INDEX_TOKEN_ID."""
     sanitized_targets = torch.where(targets != IGNORE_INDEX_TOKEN_ID, targets, tokenizer.pad_token_id)
     sanitized_predictions = torch.where(
-        predictions[PREDICTIONS] != IGNORE_INDEX_TOKEN_ID,
+        targets != IGNORE_INDEX_TOKEN_ID,
```

Comment:
@amankhandelia Thank you so much for finding the cause and proposing a fix.  Would it be possible for you as part of this pull request to include a test for the correction you made?  That would be really helpful, and hopefully would not take too much of your time.  Thanks again!